src/java/org/apache/cassandra/db/marshal/DecimalType.java

```diff
@@ -117,46 +117,56 @@ public boolean isFloatingPoint()
             return null;
         if (value.compareTo(BigDecimal.ZERO) == 0)  // Note: 0.equals(0.0) returns false!
             return ByteSource.oneByte(POSITIVE_DECIMAL_HEADER_MASK);
+
         long scale = (((long) value.scale()) - value.precision()) & ~1;
         boolean negative = value.signum() < 0;
-        final int negmul = negative ? -1 : 1;
-        // This should always fit into an int
-        final long exponent = (-scale * negmul) / 2;
+        // Make a base-100 exponent (this will always fit in an int).
+        int exponent = Math.toIntExact(-scale / 2);
```

Comment:
nit: can this division be replaced by >> 1 ?

Reply:
I was thinking of saying that the compiler will do it, but then realized it can't because the two are not the same thing (/2 rounds to 0, >>1 rounds to -inf). In this case we have already prepared even input so the two are indeed the same, but rounding to -inf is conceptually more correct. Changing.